### DISCUSSION TITLE
Display language of the code spaces editor UI (web)

### DISCUSSION THREAD

**Discussion by markusgerber76** [community/community#38946] [Codespaces] [answered]
### Select Topic Area

Product Feedback

### Body

Please allow selecting language (natural language) of the code spaces editor UI (web).  For example, many European devs prefer English as the UI language rather than French or German.

**Comment by erikaellr** [comment]
Hi @markusgerber76, we're so happy you're using Codespaces! If I'm understanding you correctly, I think we do support what you're trying to do. Press Ctrl/Cmd+Shift+P to bring up the Command Palette and type "display language" to filter, then click the "Configure Display Language" command and choose from the list of supported languages. You'll be prompted to reload for that change to take effect.

Additionally, it may help you in the future to note that the web editor (github.dev) has all the functionality of VSCode Desktop - eg, searching for "vscode change ui language" might yield useful results even where you don't find anything specific to Codespaces.

**Reply by Konamiman** [reply]
I must be missing something because "display language" yields nothing in the command palette nor in the search bar of the settings page. Neither in codespaces, nor in the desktop VS Code.

**Reply by Manbiz1** [reply]
Telegram

**Reply by wjandrea** [reply]
How do I change it for my account? I tried Settings Sync, but that didn't work. I'm not sure if I missed something or I really do need to change the display language for every new Codespace.

**Edit:** [Display language isn't synced](https://github.com/orgs/community/discussions/52603#discussioncomment-5592907) and [doesn't seem to be configurable](https://github.com/orgs/community/discussions/52603#discussion-5076207). (links go to a reply and the post it was replying to, just to be clear)

**Reply by Broodlisk** [reply]
This one solved my problem! Thank you very much.

**Reply by ethtatic** [reply]
I had the same issue that I did not find anything at first. 

You will not find this configuration option in the regular "search" bar - you need to indeed press "Ctrl/Cmd+Shift+P" to open the Command Palette which overlays on top of the search bar. Alternatively you can also type ">" in the search to access the Command Palette. 

Only in the Command Palette you can search for "display language" - even if your local language is active it will find the correct option. Once you click on it you can pick from a long list of display language options. 

Be aware: Codespaces will need to reload after the language change.

**Comment by milchalpenu** [comment]
Same. "display language"  yields nothing. 

Could you please auto select the default language selected by the developer 

- Browser language
- OS language
- Or at least Github.com web interface.

**Comment by milchalpenu** [comment]
@erikaellr 
To clarify:

I am adding a couple of screenshots.

1. Just go to github.com
3. Open some repo
4. Open in codespaces in a browser (NOT IN DESKTOP APP).


![3](https://user-images.githubusercontent.com/122907561/214831686-386bb7fe-6957-4e48-9e41-ae16897467a9.png)


then wait for the browser to load ONLINE vscode from SOMETHING..github.dev

![1](https://user-images.githubusercontent.com/122907561/214831796-9a5215ff-e2ef-4df5-9736-d29aa3af11a3.png)


![2](https://user-images.githubusercontent.com/122907561/214831812-b8e4d614-1e6c-4a02-ae5c-dbb0585e4cd8.png)


If you see the above screenshots it is in German language. 

Please give an option to make the MENUS into English.

FYI, I am using a completely ENGLISH installation of operating system, and browser.

**Reply by tfriedel** [reply]
yes please. This is infuriating! And no I don't want my browser set to english for ALL webpages.

**Reply by mootari** [reply]
[Open the command palette](https://docs.github.com/en/codespaces/reference/using-the-vs-code-command-palette-in-codespaces) and search for "Configure Display Language":
<img width="641" alt="image" src="https://github.com/community/community/assets/322014/a765a190-f209-4496-835a-1cb4337ddcaa">

After selecting the option you have a choice between various languages (with English being listed at the bottom):
<img width="622" alt="Bildschirmfoto 2024-05-19 um 01 31 47" src="https://github.com/community/community/assets/322014/51080868-830b-4ae5-a674-5856c1a14776">

**Reply by Danura30082** [reply]
@mootari 
that option dosn't exist in VScode web
![image](https://github.com/community/community/assets/135959478/1d039ba7-4fc5-4578-82e9-2c32925df955)

**Reply by mootari** [reply]
@Danura30082 Those screenshots were taken in a GitHub codespace. If the option is missing in your vscode instance then that may be an issue with your particular version or configuration.

<img width="657" alt="Screenshot 2024-05-23 at 15 24 30" src="https://github.com/community/community/assets/322014/d8022017-69e6-44ec-ac24-f624c44133e1">

**Reply by ethtatic** [reply]
I had the same issue that I did not find anything at first.

You will not find this configuration option in the regular "search" bar - you need to indeed press "Ctrl/Cmd+Shift+P" to open the Command Palette which overlays on top of the search bar. Alternatively you can also type ">" in the search to access the Command Palette.

Only in the Command Palette you can search for "display language" - even if your local language is active it will find the correct option. Once you click on it you can pick from a long list of display language options.

Be aware: Codespaces will need to reload after the language change.

**Comment by NomarCub** [comment]
My primary language in my browser is also English, and got Hungarian for Codespaces. The methods suggested here don't work. I tried `Configure Display Language` in my local VSCode install after setting it to Hungarian, and it exists there, but not on Codespaces. Normal methods don't seem to work, as the display language is not saved to settings.json.

**Comment by UsatiyNyan** [comment, marked as answer]
Fixed this in Chrome by going into settings and editing "Preferred languages" option.

**Reply by moritzfelipe** [reply]
This worked. Didn't find another way in the UI. Bad UX.

**Reply by markusgerber76** [reply]
Another option is to use this policy https://chromeenterprise.google/policies/#ForcedLanguages

**Reply by AnzhiZhang** [reply]
This is a workaround, people want other websites display as their first language, but English in Codespace.

**Comment by garysassano** [comment]
Why Codespaces inherits the language from your browser instead of your GitHub account? This makes totally no sense.

**Reply by nerkn** [reply]
because much easier to read from browser settings than connecting github, and reading from there.

**Reply by liudonghua123** [reply]
Why not add a setting in https://github.com/settings/codespaces.

**Comment by AnzhiZhang** [comment]
Please add a setting in Codespace, or synced from my account. I use my first language in browser but English in IDEs, whenever I open a new codespace, it automatically set to my browser's language.

**Comment by pior** [comment]
If Codespaces wants to be a real IDE, it needs to offer a better answer than "Change your whole browser configuration".

Setting a browser language to a local language, and sticking with English for coding/work is _very_ common.

**Comment by culudamar** [comment]
I'm using ChromeOS, language is english:
![image](https://github.com/user-attachments/assets/9bbdead9-63bc-4c39-88c8-a11bd662ddb9)

But I still see codespace UI in Turkish:
![image](https://github.com/user-attachments/assets/eb974334-7089-46f3-beb0-034ea6d03de1)
